.github/workflows/pull-request.yaml

```diff
@@ -1,3 +1,17 @@
+# Copyright 2023 Canonical Ltd.
```

Comment:
Hmm... could you explain why you have two workflow files instead of just a _ci.yaml_ file? I have seen the other charming teams do it both ways, and I prefer the approach where _ci.yaml_ runs for both pull requests and workflow calls. I could see a separate workflow being dedicated for publishing the charm when a new release is created, but I think _build-and-test.yaml_ and _pull-request.yaml_ should be combined.

Reply:
I was thinking it might be helpful where pull-requests only require a subset of jobs and build-and-test would be the big manual suite of jobs. That being said, as it exists now the split is providing much benefit so I'm okay with combing them. I'll let Omnivector determine that in the future if the case ever comes up